### PR TITLE
chore: replace badges with local static versions on docs site

### PR DESCRIPTION
**Pull request by radium-v**
# Description
Handle security concerns described in #3584:
- Generate and host static badge icons for the documentation site
  - add a `generate:badges` step to `@microsoft/site-utilities` to generate the badges
    - static badges are generated based on a config
    - dynamic badges (ex. "npm package: v.v.v") are generated by reading versions in `package.json` files for each relevant package
  - remove `placehold.it` image from `fast-foundation`'s card documentation

## Motivation & context
Subresource Integrity isn't a problem if we aren't loading files from external services.

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Comment by radium-v** [comment]
@awentzel The `site-utilities` package is where the badge generator script lives but the build-static script in `website` copies them to `static/`. The `website` project has to collect these files and ensure that they exist in `static/` so that Docusaurus can find them when it builds. Since `static/` is gitignored, the static files have to be generated in, or copied to, `static/`.

Here's an overview - in `sites/website`, the `yarn build` command:
1. runs `yarn workspace @microsoft/site-utilities generate:badges` to generate the badges
2. runs `yarn build-static`, which:
    - mutates the `fast-website` webpack config to output to the `static/` directory, then builds for production
    - **copies files from `site-utilities`, (favicon, docs site logo, and the previously-generated badges) to `static/`**
    - copies `fast-components.iife.min.js` from the `@microsoft/fast-components` to `static/` for docusaurus
3. runs `yarn prepare-docs`, which:
    - copies and modifies all of the markdown files from the main packages as well as the api-extractor
    - mutates URLs pointing to `badge.fury.io` and `img.shields.io` to point to files in the `static/` directory
4. runs `docusaurus build`
    - outputs its files to `build/`
    - copies everything present in `static/` to `build/`

(Side note: most path traversal between packages is handled by `require.resolve` to avoid having a bunch of `../` everywhere.)

**Comment by radium-v** [comment]
@awentzel I realize now that I went on a semi-unrelated tangent with my previous response - good morning!

I'm not familiar with how we currently handle files on the CDN layer for the website. Updating the PR to point to them on `static.fast.design` instead of hosting them with the bundle may only require:
- make `site-utilities` aware of the icons it's already generating, so they can be passed on to `static.fast.design`
- change the regex that modifies the badge URLs in the markdown
- eliminate the `copy` step

I don't know if this is the right move for all of the files since the badges for the "npm package: v.v.v" have to be rebuilt for each version-bumping release, and should be considered generated dynamic files.

**Comment by awentzel** [comment]
@radium-v good morning. The extra tangent was helpful so I appreciate that.  I'll approve this, then we can determine at later time implications for moving to use CDN.  Essentially, all websites should request all assets from the CDN at `https://static.fast.design`.  The bundles are not necessary, because Azure Front Door will act as a load balancer with caching of its own built-in.  Thus while the entire site does not run on a CDN, all resources are cached across different services.

File: sites/website/src/prepare-docs.js [hunk 1]
```diff
@@ -48,77 +61,53 @@ function identifyPackage(path) {
...
+        const dest = path.join(__dirname, "../docs", docsFolderDest, filename);
 
         await safeCopy(source, dest);
     }
 }
 
 async function copyArticleMarkdown() {
     await moveMarkdownFiles(
-        "../../packages/web-components/fast-foundation/docs/integrations",
+        path.resolve(fastFoundation, "docs/integrations"),
```
Comment (EisenbergEffect) [inline sites/website/src/prepare-docs.js:92]: Yeah...seems like I forgot about `path.resolve` when originally whipping some of this up.... 🤣

**Pull request merged**